### PR TITLE
Chore(Tests): CNX-8640 core integration test project

### PR DESCRIPTION
**Pull request by JR-Morgan**
- Resolved naming violations in Integration test projects
- Updated editor config to warn when using the old nunit functions
- Updated both Unit and integration tests to use the new unit functions
- Removed ConfigureAwaits from test project:
    - My reasoning is. They aren't needed, don't provide value, just adds visual clutter.
    - Happy to discuss and revert if prefered.

### CODE REVIEW

File: Core/Tests/Speckle.Core.Tests.Unit/Api/GraphQLClient.cs [hunk 1]
```diff
@@ -64,9 +64,7 @@ public void TestExceptionThrowingFromGraphQLErrors(Type exType, Map extensions)
   [Test]
   public void TestMaybeThrowsDoesntThrowForNoErrors()
   {
-    _client.MaybeThrowFromGraphQLErrors(new GraphQLRequest(), new GraphQLResponse<string>());
-    // We're just checking that the prev function didn't throw
-    Assert.True(true);
```
Comment (AlanRynne) [inline Core/Tests/Speckle.Core.Tests.Unit/Api/GraphQLClient.cs:69]: Ohh WTH 🤣

File: Core/Tests/Speckle.Core.Tests.Unit/Api/Operations/OperationsReceiveTests.Exceptional.cs [hunk 2]
```diff
@@ -33,7 +33,7 @@ public void Receive_OperationCanceled_ExceptionThrown(string id)
     ctc.Cancel();
 
     MemoryTransport emptyTransport2 = new();
-    Assert.ThrowsAsync<OperationCanceledException>(async () =>
+    Assert.CatchAsync<OperationCanceledException>(async () =>
```
Comment (AlanRynne) [inline Core/Tests/Speckle.Core.Tests.Unit/Api/Operations/OperationsReceiveTests.Exceptional.cs:36]: Is `CatchAsync` the recommended pattern to assert an exception type here? Asking out of ignorance

**Review by AlanRynne — APPROVED** [review]
No complaints, just a question 👇🏼 Approving anyways

File: Core/Tests/Speckle.Core.Tests.Unit/Api/Operations/OperationsReceiveTests.Exceptional.cs (hunk 2 above)
Comment (JR-Morgan) [reply Core/Tests/Speckle.Core.Tests.Unit/Api/Operations/OperationsReceiveTests.Exceptional.cs:36]: ThrowsAsync tests if the method throws the EXACT type
CatchAsync allows subtypes

In this case, the spirit of the test was to make sure it an `OperationCancelledException`, we don't care if its a subtype like `TaskCancelledException`, so I after learning of the existence of `CatchAsync`, I decided to update the functions where we would be ok with subtypes.

**Pull request merged**